src/App.tsx

```diff
@@ -31,11 +31,18 @@ function AppRoot() {
     setShowCreateVFEForm(false);
   }
 
+  function handleUpdateVFE(updatedVFE: VFE) {
+    console.log("Updated VFE (in AppRoot):", updatedVFE);
+    setVFEData(updatedVFE);
+  }
+
   function renderComponent() {
     if (showCreateVFEForm) {
       return <CreateVFEForm onCreateVFE={loadCreatedVFE} />;
     } else if (vfeData && showApp) {
-      return <PhotosphereEditor vfe={vfeData} />;
+      return (
+        <PhotosphereEditor initialVFE={vfeData} onUpdateVFE={handleUpdateVFE} />
```

Comment:
Maybe we should leave the name vfe instead of initialVFE since it will no longer be the initial vfe after an update?

Reply:
I think the idea is that initialVFE is a property passed down from the parent.  I do think there should be a qualifier to communicate that it's not just a local variable. I'm wondering if vfe is too general.  Maybe parentVFE?